packages/babel-parser/src/tokenizer/index.js

```diff
@@ -1035,8 +1035,11 @@ export default class Tokenizer extends LocationParser {
       next = this.input.charCodeAt(this.state.pos);
     }
 
-    // disallow numeric separators in non octal decimals
-    if (this.hasPlugin("numericSeparator") && isNonOctalDecimalInt) {
+    // disallow numeric separators in non octal decimals and legacy octal likes
+    if (
+      (this.hasPlugin("numericSeparator") && isNonOctalDecimalInt) ||
+      (this.hasPlugin("numericSeparator") && octal)
```

Comment:
Can we check `this.hasPlugin` just once?

```js
this.hasPlugin("numericSeparator") && (octal || isNonOctalDecimalInt)
```

Reply:
Oh sure! That's better, thank you!